src/dp_abstraction/aca_dataplane_ovs.cpp

```diff
@@ -354,121 +361,138 @@ int ACA_Dataplane_OVS::update_neighbor_state_workitem(NeighborState current_Neig
-                  aca_get_neighbor_type_string(
-                          current_NeighborConfiguration.fixed_ips(0).neighbor_type()),
-                  current_NeighborConfiguration.project_id().c_str(),
-                  current_NeighborConfiguration.vpc_id().c_str(), found_network_type,
-                  virtual_ip_address.c_str(), virtual_mac_address.c_str(),
-                  host_ip_address.c_str(), found_tunnel_id);
-
-          // with Alcor DVR, a cross subnet packet will be routed to the destination subnet.
-          // that means a L3 neighbor will become a L2 neighbor, therefore, call the below
-          // for both L2 and L3 neighbor update
-
-          // only need to update L2 neighbor info if it is not on the same compute host
-          bool is_neighbor_port_on_same_host = aca_is_port_on_same_host(host_ip_address);
-
-          if (is_neighbor_port_on_same_host) {
-            ACA_LOG_DEBUG("neighbor host: %s is on the same compute node, don't need to update L2 neighbor info.\n",
-                          host_ip_address.c_str());
-            overall_rc = EXIT_SUCCESS;
+          if (!subnet_info_found) {
+            ACA_LOG_ERROR("Not able to find the info for neighbor ip_index: %d with subnet ID: %s.\n",
+                          ip_index,
+                          current_NeighborConfiguration.fixed_ips(ip_index)
+                                  .subnet_id()
+                                  .c_str());
+            overall_rc = -EXIT_FAILURE;
           } else {
-            overall_rc = ACA_OVS_L2_Programmer::get_instance().create_update_neighbor_port(
-                    current_NeighborConfiguration.vpc_id(), found_network_type, host_ip_address,
-                    found_tunnel_id, culminative_dataplane_programming_time);
-            // we can consider doing this L2 neighbor creation as an on demand rule to support scale
-            // when we are ready to put the DVR rule as on demand, we should put the L2 neighbor rule
-            // as on demand also
-          }
-
-          if (overall_rc == EXIT_SUCCESS) {
-            if (current_NeighborConfiguration.fixed_ips(0).neighbor_type() ==
-                NeighborType::L3) {
-              overall_rc = ACA_OVS_L3_Programmer::get_instance().create_neighbor_l3(
-                      current_NeighborConfiguration.vpc_id(),
-                      current_NeighborConfiguration.fixed_ips(0).subnet_id(), found_network_type,
-                      virtual_ip_address, virtual_mac_address, host_ip_address,
+            ACA_LOG_DEBUG(
+                    "Neighbor Operation:%s: neighbor_type:%s, project_id:%s, vpc_id:%s, network_type:%d, ip_index: %d,"
+                    "virtual_ip_address:%s, virtual_mac_address:%s, neighbor_host_ip_address:%s, tunnel_id:%d\n",
+                    aca_get_operation_string(current_NeighborState.operation_type()),
+                    aca_get_neighbor_type_string(
```

Comment:
Is it possible to throw an exception in this long aca_get_neighbor_type_string?

Reply:
no, it is just print error when the input is invalid :)